src/rdb/rdb.c

```diff
@@ -503,17 +518,32 @@ rdb_add_replicas(struct rdb *db, d_rank_list_t *replicas)
 int
 rdb_remove_replicas(struct rdb *db, d_rank_list_t *replicas)
 {
-	int i;
-	int rc = -DER_INVAL;
+	int	i;
+	int	rc;
 
 	D_DEBUG(DB_MD, DF_DB": Removing %d replicas\n",
 		DP_DB(db), replicas->rl_nr);
+
+	ABT_mutex_lock(db->d_raft_mutex);
```

Comment:
Even without the addition of the rdb_raft_wait_applied() call in this patch, do you think that taking the lock here only once - around the whole replicas->rl_nr loop (rdb_raft_remove_replica calls) is better than the previous code that locks/unlocks for each replica?

It seems like a good change from that perspective. IIRC when the original locking code was added there was a desire to scope the locking as narrowly as possible around entry / return from the raft state machine. So it ended up being done more narrowly around the calls to  rdb_raft_append_apply_internal().

Reply:
The difference in locking doesn't matter too much, I think. rdb_raft_wait_applied releases the lock during the CV wait(s). And we don't expect other pthreads to access the same rdb object. Lastly, considering we are doing membership changes, rather than appending regular entries, we might want to be conservative wrt concurrency---until some future, unexpected issue proves otherwise, that is.